rmw_zenoh_cpp/src/detail/rmw_data_types.hpp

```diff
@@ -121,14 +122,27 @@ struct rmw_subscription_data_t
   MessageTypeSupport * type_support;
   rmw_context_t * context;
 
-  std::deque<std::unique_ptr<saved_msg_data>> message_queue;
-  std::mutex message_queue_mutex;
-
   size_t queue_depth;
   bool reliable;
 
-  std::mutex internal_mutex;
-  std::condition_variable * condition{nullptr};
+  void attach_condition(std::condition_variable * condition_variable);
```

Comment:
How do you feel about defining a separate concrete class with the condition variable and mutex and protected members, and the `attach_condition()` and `detach_condition()` methods defined? Then each of `rmf_X_data_t` classes will inherit this class. It will reduce code duplication a bit.

Reply:
While I totally understand the desire to do so, in this case I don't think it is a good idea.  As it stands, these are just "fancy structs" that have some methods attached to them.  As soon as you add in inheritance, then you add in complications with polymorphism, and also probably slower code due to the addition of vtables.  The latter can probably be mitigated with some clever definition of destructors, but the amount of duplicated code here is minimal, so I'd rather just duplicate it.

Reply:
Sounds good, thanks for clarifying.